src/Compiler.ts

```diff
@@ -113,32 +132,41 @@ function removeInvalidOptionsFromArgv(commandLine: ReadonlyArray<string>, invali
 // Polyfill for report diagnostics
 function logError(commandLine: ts.ParsedCommandLine) {
     if (commandLine.errors.length !== 0) {
+        let invalidErrorCount = 0;
         commandLine.errors.forEach((err) => {
-            console.log(err.messageText);
+            // Allow all compiler options
+            if (err.code == 5023) {
+                invalidErrorCount += 1;
+            }
+            else {
+                console.log(err.messageText);
+            }
         });
-        process.exit(1);
+        if (invalidErrorCount != commandLine.errors.length) {
+            process.exit(1);
+        }
```

Comment:
all of `logError` looks pretty dodgy. Improve or add some comments that explain what/why this function does.